src/GapicClientTrait.php

```diff
@@ -674,9 +684,12 @@ private function configureCallConstructionOptions(string $methodName, array $opt
                 );
             }
         }
-        return [
-            'retrySettings' => $retrySettings,
-        ];
+        $result = ['retrySettings' => $retrySettings];
+
+        if (!empty($autoPopulatedFields)) {
+            $result['autoPopulationSettings'] = $autoPopulatedFields;
+        }
+        return $result;
```

Comment:
this seems complicated, why not just 
```suggestion
        return [
            'retrySettings' => $retrySettings,
            'autoPopulationSettings' => $autoPopulatedFields,
        ];
```
I don't think there's an advantage between an empty `autoPopulationSettings` and not setting it at all

Reply:
Doing this would break the `testConfigureCallConstructionOptionsAcceptsRetryObjectOrArray` [test]( https://github.com/googleapis/gax-php/blob/main/tests/Tests/Unit/GapicClientTraitTest.php#L138-L163). The fix would be to just add the `"autoPopulationSettings" => []` entry in [expected options](https://github.com/googleapis/gax-php/blob/main/tests/Tests/Unit/GapicClientTraitTest.php#L143-L146).

Should we do that instead?

Reply:
Yes, we shouldn't write code to match existing tests, but rather update the tests when we make the desired changes to the code. 

LGTM